app/forms/questionnaire_form.py

```diff
@@ -176,11 +176,25 @@ def _get_target_total_and_currency(
         question: QuestionSchema,
     ) -> Optional[tuple[Union[Calculation, AnswerValueTypes], Optional[str]]]:
 
+        list_item_id = self.location.list_item_id if self.location else None
+        value_source_resolver = ValueSourceResolver(
+            answer_store=self.answer_store,
+            schema=self.schema,
+            metadata=self.metadata,
+            response_metadata=self.response_metadata,
+            list_store=self.list_store,
+            location=self.location,
+            list_item_id=list_item_id,
+        )
```

Comment:
I would make this a property of the form. You only need one instance of this per form instance.

Reply:
Ok, top property of the form now, cheers!